observer/observer_test.go

```diff
@@ -0,0 +1,141 @@
+package observer
```

Comment:
In Go tests should be in the special `_test` package, so they are not exposed along with the root package
```suggestion
package observer_test
```